cmd/setProperties.go

```diff
@@ -0,0 +1,120 @@
+// Copyright © 2017 Microsoft <wastore@microsoft.com>
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/Azure/azure-storage-azcopy/v10/common"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+func (raw *rawCopyCmdArgs) setMandatoryDefaultsForSetProperties() {
+	raw.blobType = common.EBlobType.Detect().String()
+	//raw.blockBlobTier = common.EBlockBlobTier.None().String()
+	//raw.pageBlobTier = common.EPageBlobTier.None().String()
+	raw.md5ValidationOption = common.DefaultHashValidationOption.String()
+	raw.s2sInvalidMetadataHandleOption = common.DefaultInvalidMetadataHandleOption.String()
+	raw.forceWrite = common.EOverwriteOption.True().String()
+	raw.preserveOwner = common.PreserveOwnerDefault
+}
+
+func init() {
+	raw := rawCopyCmdArgs{}
+
+	setPropCmd := &cobra.Command{
+		Use:        "set-properties",
+		Aliases:    []string{},
+		SuggestFor: []string{},
+		// TODO: t-iverma: short and long descriptions
+		Short: "Change properties of given blob",
+		Long:  "",
+		Args: func(cmd *cobra.Command, args []string) error {
+			// we only want one arg, which is the source
+			if len(args) != 1 {
+				return fmt.Errorf("setproperties command only takes 1 arguments (src). Passed %d arguments", len(args))
+			}
+
+			//the resource to set properties of is set as src
+			raw.src = args[0]
+
+			//TODO what is the purpose of these lines?
+			srcLocationType := InferArgumentLocation(raw.src)
+			if raw.fromTo == "" {
+				switch srcLocationType {
+				case common.ELocation.Blob():
+					raw.fromTo = common.EFromTo.BlobNone().String()
+				case common.ELocation.BlobFS():
+					raw.fromTo = common.EFromTo.BlobFSTrash().String()
+				default:
+					return fmt.Errorf("invalid source type %s to delete. azcopy support removing blobs/files/adls gen2", srcLocationType.String())
+				}
+			} else if raw.fromTo != "" {
+				err := strings.Contains(raw.fromTo, "Trash")
```

Comment:
Changed, strings will pass this preliminary check if you have "None" in them. Eg FileNone. However, some strings like "FileNonexyz" will pass too. This will be handled by the cook method